### PR TITLE
[clang][cas] Remove unsafe global CAS object from clang-scan-deps

### PR DESCRIPTION
**Pull request by benlangmuir**
When using a plugin CAS, the global destruction may cause crashes if we have a global CAS object.

rdar://116235144

### CODE REVIEW

**Comment by benlangmuir** [comment]
@swift-ci please test llvm

**Review by akyrtzi — APPROVED** [review]

File: clang/tools/clang-scan-deps/ClangScanDeps.cpp [hunk 1]
```diff
@@ -220,12 +223,12 @@ static void ParseArgs(int argc, char **argv) {
   InMemoryCAS = Args.hasArg(OPT_in_memory_cas);
 
   if (const llvm::opt::Arg *A = Args.getLastArg(OPT_cas_path_EQ))
-    CASOpts.CASPath = A->getValue();
+    CASConfig.CASPath = A->getValue();
```
Comment (cachemeifyoucan) [inline clang/tools/clang-scan-deps/ClangScanDeps.cpp]: Here, we can just parse into a `static std::string CASPath`, then construct the CASOptions in the actual invocation?

Comment (cachemeifyoucan) [reply clang/tools/clang-scan-deps/ClangScanDeps.cpp]: Then we can simply follow the rule that word `CAS` should not appear in global/static variables types.

**Comment by benlangmuir** [comment]
@swift-ci please test llvm

File: clang/tools/clang-scan-deps/ClangScanDeps.cpp (hunk 1 above)
Comment (benlangmuir) [reply clang/tools/clang-scan-deps/ClangScanDeps.cpp]: Done in 033c965

**Review by cachemeifyoucan — APPROVED** [review]
Thanks. LGTM.

**Pull request merged**